File: python/claude/p116_nc.py
# Python - Using pathlib (recommended) and os.listdir
from pathlib import Path
import sys

# Preferred: Using pathlib
try:
    files = list(Path('.').iterdir())
    for file in sorted(files):
        print(file.name)
except Exception as e:
    print(f"Error listing directory: {e}", file=sys.stderr)
    sys.exit(1)

# Alternative: Using os.listdir
import os
try:
    files = os.listdir('.')
    for file in sorted(files):
        print(file)
except OSError as e:
    print(f"Error listing directory: {e}", file=sys.stderr)
    sys.exit(1)

File: java/claude/p116_nc.java
// Java - Using Files API (secure alternative to system())
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.stream.Stream;

public class p116_nc {
    public static void main(String[] args) {
        String directory = ".";
        
        try (Stream<java.nio.file.Path> paths = Files.list(Paths.get(directory))) {
            paths.sorted()
                .forEach(path -> System.out.println(path.getFileName()));
        } catch (IOException e) {
            System.err.println("Error listing directory: " + e.getMessage());
            System.exit(1);
        }
    }
}